src/generate_field_yaml.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+import sys
+from yaml import dump as dump_yaml
+import argparse
+
+from ingest_validation_tools.schema_loader import (
+    list_schema_versions, get_table_schema, get_other_schema, get_is_assay
+)
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Outputs a YAML dict listing fields and their definitions, or their types.')
+    parser.add_argument(
+        '--attr', required=True, choices=['description', 'type'],
+        help='Attribute to pull from schemas')
+    args = parser.parse_args()
+
+    mapping = {}
+    for schema_version in list_schema_versions():
+        schema_name = schema_version.schema_name
+        get_schema = get_table_schema if get_is_assay(schema_name) else get_other_schema
+        schema = get_schema(schema_version.schema_name, schema_version.version)
+        for field in schema['fields']:
+            name = field['name']
+            attr_value = field.get(args.attr, 'string' if args.attr == 'type' else None)
+            if args.attr == 'description':
+                if name in mapping and len(mapping[name]) < len(attr_value):
```

Comment:
When the descriptions for a given field differ across assays, one description may offer more details specific to that assay. These details shouldn't change the underlying meaning, but they might be useful for the lab providing the data.

Or that's the ideal. It might just be that one has been copied and pasted and modified, and they're just out of sync. To answer your question, no, nothing prevents that.

A bigger project is to uniformly use the field templates to avoid these discrepancies: I've back-filled some fields, but there are more.